crates/ruff_python_formatter/tests/normalizer.rs

```diff
@@ -81,7 +81,7 @@ impl Transformer for Normalizer {
                         string.value = ast::StringLiteralValue::single(ast::StringLiteral {
                             value: string.value.to_str().to_string().into_boxed_str(),
                             range: string.range,
-                            flags: StringLiteralFlags::default(),
+                            flags: string.value.flags(),
```

Comment:
I'd still love somebody more familiar with the formatter than me to OK this change -- I know Micha is out this week, so maybe @dhruvmanila?

Reply:
tl;dr, I don't think this matters but I'd prefer to keep using the default flags.

This normalizer is mainly used to normalize the formatted and unformatted AST to assert that both the ASTs are same (formatter shouldn't change the AST). This assertion happens here: https://github.com/astral-sh/ruff/blob/648ad8a20de2bc709470ffe908e42687d87dedc7/crates/ruff_python_formatter/tests/fixtures.rs#L384-L422

Why I think this doesn't matter is because after the normalization, we compare the AST using the comparable logic which only compares the string by its content and not any of the flags:

https://github.com/astral-sh/ruff/blob/648ad8a20de2bc709470ffe908e42687d87dedc7/crates/ruff_python_ast/src/comparable.rs#L694-L697

Reply:
Cool. So this + our other conversation at https://github.com/astral-sh/ruff/pull/15726#discussion_r1930545195 suggests to me that we should:
- Get rid of `StringLiteralFlags::new()`
- Promote `StringLiteralFlags::empty()` to `pub` rather than `pub(crate)` (but add a doc-comment that warns that you probably don't really ever want to use `StringLiteralFlags::empty()` from the linter crate)
- Use `StringLiteralFlags::empty()` at this location in the formatter